includes/Core/Authentication/Google_Proxy.php

```diff
@@ -399,24 +450,59 @@ public function unregister_site( Credentials $credentials ) {
 		return $this->request( self::OAUTH2_DELETE_SITE_URI, $credentials );
 	}
 
+	/**
+	 * Registers the site on the proxy.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string|WP_Error Redirect URL on success, otherwise an error.
+	 */
+	public function register_site() {
+		return $this->sync_site_fields( null, null );
```

Comment:
See my other comment, maybe better to use a private "low-level" method here. Also, let's either also add the `$mode = 'async'` parameter so that it can be controlled from the outside, or we need to specify `sync` here because the only use-case we have for it now needs to be `sync`.

For parity with the other method, I prefer having a parameter for it that can be controlled from the outside.

Reply:
Updated.